src/util/address.rs

```diff
@@ -714,9 +722,15 @@ impl FromStr for Address {
                 return Err(Error::InvalidWitnessProgramLength(program.len()));
             }
 
-            // Specific segwit v0 check.
-            if version == WitnessVersion::V0 && (program.len() != 20 && program.len() != 32) {
-                return Err(Error::InvalidSegwitV0ProgramLength(program.len()));
+            // Specific witness version checks.
+            match (version, program.len()) {
+                (WitnessVersion::V0, len) if len != 20 && len != 32 => {
+                    return Err(Error::InvalidSegwitV0ProgramLength(len))
+                },
+                (WitnessVersion::V1, len) if len != 32 => {
```

Comment:
> Any other outputs, including version 1 outputs with lengths other than 32 bytes, or P2SH-wrapped version 1 outputs[3], remain unencumbered.

As far as I can recollect, this is not correct. We can have (I hope we never do) a new `(WitnessVersion::V1, 33)` in the future.

Reply:
I really want to return this error here, but I think that would be wrong.

Reply:
No, since non-32 byte V1 will be a hard fork

Reply:
> Any other outputs, including version 1 outputs with lengths other than 32 bytes, or P2SH-wrapped version 1 outputs[3], remain unencumbered.

Sorry, I should have highlighted that this statement is from BIP341. And I have verified with the implementation, that non-32 byte V1 won't be a hard fork. I hope never to see that soft fork, but it is what it is.

Reply:
Wow, thank you for spotting that! Are you aware of the rationale for a such weird thing?

Reply:
I guess it's left for possible future upgrades in case we run out of version numbers or something.